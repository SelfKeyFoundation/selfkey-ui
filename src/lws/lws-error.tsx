import * as React from 'react';
import injectSheet, { StyleSheet } from 'react-jss';
import CommonStyle from '../common/common-style';

const styles: StyleSheet = {
	form: CommonStyle.form,

	formSubmitRow: {
		marginTop: '15px',
		'& button': {
			marginTop: '30px',
		},
	},

	supportText: {
		textAlign: 'center',
		lineHeight: '22px',
		fontFamily: 'ProximaNovaSemibold, arial, sans-serif',
		color: '#C5DCE9',
		padding: '0 0 30px',
	},

	areaTitle: {
		textAlign: 'center',
		'& h2': {
			textTransform: 'uppercase',
			fontWeight: 'normal',
			fontSize: '16px',
			padding: '25px 0',
			margin: 0,
			color: '#D97300',
		},
		'& h3': {
			fontWeight: 'normal',
			fontSize: '21px',
			padding: '0 0 45px',
			margin: 0,
			color: '#93B0C1',
		},
	},
};

export type LWSErrorProps = {
	actionButton?: React.ReactElement<any>;
	actionName: string;
	errorName: string;
	actionIcon: React.ReactElement<any>;
	supportText: string;
};

export const LWSError = injectSheet(styles)<LWSErrorProps>(
	({ classes, actionButton, actionName, errorName, actionIcon, supportText }) => (
		<div>
			<div className={classes.areaTitle}>
				{actionIcon}
				<h2>{actionName}</h2>
				<h3>{errorName}</h3>
			</div>
			<div className={classes.form}>
				<p className={classes.supportText}>{supportText}</p>
				{actionButton ? <div className={classes.formSubmitRow}>{actionButton}</div> : null}
			</div>
		</div>
	)
);

export default LWSError;
