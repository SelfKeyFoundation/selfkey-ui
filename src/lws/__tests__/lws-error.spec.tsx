import * as React from 'react';
import { create } from 'react-test-renderer';
import { LWSError } from '../lws-error';

describe('LWS Error Screen', () => {
	it('should show LWS Error Screen', () => {
		const component = create(
			<LWSError
				actionName="test"
				errorName="test"
				supportText="test"
				actionButton={<button>retry</button>}
				actionIcon={<span>icon</span>}
			/>
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
