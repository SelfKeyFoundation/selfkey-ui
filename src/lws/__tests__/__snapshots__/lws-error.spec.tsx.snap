// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LWS Error Screen should show LWS Error Screen 1`] = `
<div
  className="Component-lwsError-0-2-2"
>
  <div
    className="Component-areaTitle-0-2-5"
  >
    <span>
      icon
    </span>
    <h2>
      test
    </h2>
    <h3>
      test
    </h3>
  </div>
  <div
    className="Component-form-0-2-1"
  >
    <p
      className="Component-supportText-0-2-4"
    >
      test
    </p>
    <div
      className="Component-formSubmitRow-0-2-3"
    >
      <button>
        retry
      </button>
    </div>
  </div>
</div>
`;
